test/replication-luatest/gh_10766_shutdown_non_empty_limbo_test.lua

```diff
@@ -0,0 +1,110 @@
+local t = require('luatest')
```

Comment:
The commit msg says:

> - The txn was synchro and being written to WAL right now. Then it
      was also freed (including its region memory removal), and
      the WAL thread would sometimes crash on use-after-free.

However, without the patch I didn't manage to reproduce the crash. In the majority of cases the test passes. The first test always passes, the second one sometimes hangs. I was testing on debug build. Am I doing smth wrong or it's reproduced very rarely? If it's the latter, are we ok with so rarely reproducible test?

Reply:
I tried to catch it in Ubuntu (in docker) x86, Ubuntu ARM (also docker), Mac M3 - only the last one crashes reliably.

But I tried to improve it. I've moved the WAL delay a bit earlier, so after it is released, it will try to access more of the region data. On Mac both the test cases crash without the patch. Can you try again on your machine?

Reply:
Yep, it's fails reliably now, thank you